samza-core/src/main/java/org/apache/samza/operators/MessageStreamImpl.java

```diff
@@ -163,10 +163,11 @@ public K getOtherKey(M message) {
 
   @Override
   public <K> MessageStream<M> partitionBy(Function<M, K> parKeyExtractor) {
-    MessageStreamImpl<M> intStream = this.graph.createIntStream(parKeyExtractor);
+    int opId = graph.getNextOpId();
+    MessageStreamImpl<M> intStream = this.graph.generateIntStreamFromOpId(opId, parKeyExtractor);
     OutputStream<M> outputStream = this.graph.getOutputStream(intStream);
     this.registeredOperatorSpecs.add(OperatorSpecs.createPartitionOperatorSpec(outputStream.getSinkFunction(),
-        this.graph, outputStream));
+        this.graph, outputStream, opId));
```

Comment:
Makes sense. I removed the graph in this function. Let's sync up with @nickpan47 about your second suggestion.